### PR TITLE
build(deps): Upgrade setup-go action to use go.mod as a version file

### PR DESCRIPTION
**Pull request by abtris**
Signed-off-by: Ladislav Prskavec <ladislav@prskavec.net>

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -16,21 +16,20 @@ jobs:
...
       - name: Install Go
-        uses: actions/setup-go@v3.0.0
+        uses: actions/setup-go@v3.1.0
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version-file: "go.mod"
 
       - name: Checkout keptn-contrib/keptn-service-template-go
-        uses: actions/checkout@v3
+        uses: actions/checkout@v3.0.2
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml:32]: Here we still need to move the checkout before setup-go

Comment (abtris) [reply .github/workflows/integration_tests.yml:27]: Fixed

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Comment by Raffy23** [comment]
Hi @abtris, it seems there are some merge conflicts that must be resolved before merging. After that, we can approve the PR again so that you can merge it into main.

**Comment by abtris** [comment]
@Raffy23 rebased with current main

**Comment by abtris** [comment]
@Raffy23 I don't have the right to merge it.

**Comment by Raffy23** [comment]
Sorry, I wasn't aware that you can't merge the request, I will merge it for you.

**Pull request merged**